src/Storages/AlterCommands.cpp

```diff
@@ -1069,6 +1070,13 @@ void AlterCommands::validate(const StoragePtr & table, ContextPtr context) const
         const auto & column_name = command.column_name;
         if (command.type == AlterCommand::ADD_COLUMN)
         {
+            /// FIXME: Adding a new column of type Object(JSON) is broken.
+            /// Looks like there is something around default expression for this column (method `getDefault` is not implemented for the data type Object).
+            /// But after ALTER TABLE ADD COLUMN we need to fill existing rows with something (exactly the default value).
+            /// So we don't allow to do it for now.
+            if (command.data_type && command.data_type->hasDynamicSubcolumns())
```

Comment:
How is it possible that it doesn't have data type?

Reply:
Ah, it's not: https://github.com/ClickHouse/ClickHouse/blob/64b38ab7e5f0a3a44f7e8d9f55ba71953fb1c683/src/Storages/AlterCommands.cpp#L1090-L1092

Reply:
Looks like it is possible: `ALTER TABLE table2 ADD COLUMN Value2 DEFAULT 'some_string';`